src/server.js

```diff
@@ -432,17 +446,18 @@ module.exports = function (classes) {
               return;
             }
 
-            resolve(decoded);
+            return resolve(decoded);
           };
 
           socket.on('message', function (event) {
             try {
               parser.write(event.data);
             } catch (err) {
               // TODO: Is ignoring invalid data the right thing to do?
+              return reject(new Error.InvalidParams(err.message));  
```

Comment:
the problem with this is that if the JSON is malformed, it will leak internal data to the client. security wise it's not a good sign